internal/ingress/controller/nginx_test.go

```diff
@@ -147,6 +148,27 @@ func TestIsDynamicConfigurationEnough(t *testing.T) {
 }
 
 func TestConfigureDynamically(t *testing.T) {
+	l, err := net.Listen("unix", nginxStreamSocket)
+	if err != nil {
+		t.Fatalf("unexpected error creating unix socket: %v", err)
+	}
+	if l == nil {
+		t.Fatalf("expected a listener but none returned")
+	}
+	defer l.Close()
+
+	go func() {
+		for {
+			conn, err := l.Accept()
+			if err != nil {
+				continue
+			}
+
+			time.Sleep(100 * time.Millisecond)
+			defer conn.Close()
+		}
+	}()
+
```

Comment:
This is just a mock right? If so can you extract this into a function that describes it?

Reply:
done